packages/enzyme/src/selectors.js

```diff
@@ -354,11 +354,20 @@ function matchDescendant(nodes, predicate) {
  * the selector. The selector can be a simple selector, which
  * is handled by `buildPredicate`, or a complex CSS selector which
  * reduceTreeBySelector parses and reduces the tree based on the combinators.
- * @param {Function|Object|String} selector
+ *
+ * Logic to check if selector is a HTMLElement referenced from
+ * https://stackoverflow.com/questions/384286/javascript-isdom-how-do-you-check-if-a-javascript-object-is-a-dom-object
+ *
+ * @param {EnzymeSelector} selector
  * @param {RSTNode} root
+ * //TODO Keith
  */
 export function reduceTreeBySelector(selector, root) {
-  if (typeof selector === 'function' || typeof selector === 'object') {
+  // Check if the selector is a HTMLElement and assign it to a variable
+  const isElement = (typeof HTMLElement === 'object') ? selector instanceof HTMLElement : 
```

Comment:
I'd prefer to find a detection method that's not `instanceof`, since this doesn't work cross-realm, if possible.
```suggestion
  const isElement = typeof HTMLElement === 'object' ? selector instanceof HTMLElement : 
```